### PR TITLE
Clang Compatibility

### PR DESCRIPTION
**Pull request by tete17**
Pull Request to correct specific clang incompatibilities.
Fixes issue #320 

Credit to this 2 stack overflow questions:
 - https://stackoverflow.com/questions/7411515/why-does-c-require-a-user-provided-default-constructor-to-default-construct-a
 - https://stackoverflow.com/questions/14197436/c11-template-parameter-redefines-default-argument

### CODE REVIEW

**Comment by jdgarciauc3m** [comment]
Good approach!

However, the pull-request is incomplete. We need to apply it also to member function declarations, not only definitions.

Additionally this should be applied to all execution policies.

File: CMakeLists.txt [hunk 1]
```diff
@@ -77,7 +77,7 @@ if( GRPPI_OMP_ENABLE )
 endif( GRPPI_OMP_ENABLE )
 
 #FastFlow library
-option( GRPPI_FF_ENABLE "Require FastFlow library" ON )
+option( GRPPI_FF_ENABLE "Require FastFlow library" OFF )
```
Comment (jdgarciauc3m) [inline CMakeLists.txt]: This change is not related to this issue. We should leave it to ON.

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (tete17) [reply CMakeLists.txt]: That is right it slipped into my commit accidentally.

**Comment by tete17** [comment]
Regarding the declarations, there still need to be default parameters.

When we call one of such functions all "normal" parameters get deduced but there "require like" parameters require to be assign to something, even a dumb value such as 0 otherwise the compiler will complain and not compile when called.

That is why we need somewhere a default parameters, the issue is that it is only allowed to be either in the declaration or definitions.   

Please let me know if i understood your request wrongly
Looking forward to collaborating with your team

**Review by jdgarciauc3m — APPROVED** [review]
The PR correctly addresses the issues with clang.

Thank you very much for your contribution.

**Pull request merged**